Kernel/Arch/i386/CPU.cpp

```diff
@@ -829,6 +829,22 @@ FlatPtr read_dr6()
     return dr6;
 }
 
+#define XCR_XFEATURE_ENABLED_MASK 0
+
+u64 read_xcr0()
+{
+    u32 eax, edx;
+    asm volatile("xgetbv" : "=a"(eax), "=d"(edx) : "c" (XCR_XFEATURE_ENABLED_MASK));
+    return eax + ((u64)edx << 32);
+}
+
+void write_xcr0(u64 value)
+{
+    u32 eax = value;
+    u32 edx = value >> 32;
+    asm volatile("xsetbv" :: "a"(eax), "d" (edx), "c" (XCR_XFEATURE_ENABLED_MASK));
+}
```

Comment:
Since neither of these functions are needed after the kernel has booted into userspace, please mark them as `UNMAP_AFTER_INIT` :)

Reply:
Thanks, updated the PR.